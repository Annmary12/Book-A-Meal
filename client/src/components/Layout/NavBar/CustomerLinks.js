import React, { Fragment } from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import styles from './navbar.scss';

const CustomerLinks = ({ pathname }) => (
  <Fragment>
    <li>
      <Link
        className={classNames({
          [styles.active]: pathname === '/menus'
        })}
        to="/menus"
      >
        Menus
      </Link>
    </li>
    <li>
      <Link
        className={classNames({
          [styles.active]: pathname === '/orders'
        })}
        to="/orders"
      >
        Order history
      </Link>
    </li>
  </Fragment>
);

export default CustomerLinks;
