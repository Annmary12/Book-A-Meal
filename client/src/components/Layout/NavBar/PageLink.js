import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './navbar.scss';

const PageLink = ({ pathname, to, text }) => (
  <Link
    className={classNames({
      [styles.active]: pathname === to
    })}
    to={to}
  >
    {text}
  </Link>
);

PageLink.propTypes = {
  pathname: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default PageLink;
