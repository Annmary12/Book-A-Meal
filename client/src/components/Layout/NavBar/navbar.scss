.nav{
  background-color: #d24e24;
  padding: 9px;
  color: white;

  @media only screen and (max-width: 600px){
    position: absolute;
    padding: 9px;
    color: white;
    margin-bottom: 0px;
    width: 96%;
  }


  @media only screen and (min-width: 600px) {
    display: flex;
    flex-direction: row;
    padding-left: 30px;
    padding-right: 30px;

    /*background-color: #a82800;*/
    margin-bottom: 0px;
    height: 40px;

  }
}

.titleArea{
  margin-top: 4px;
  @media only screen and (max-width: 600px){
    display: flex;
		width: 100%;
  }
}

.title{
	font-family: "Dancing Script",cursive, calibri;
	font-size: 30px;
  font-weight: bold;
  position: absolute;
  left: 69px;
  text-decoration: none;
  color: white;
  text-align: center;

  @media only screen and (max-width: 600px){
    position: absolute;
    top:none;
    left: none;
    text-decoration: none;
    color: white;
  }
}

.logo{
  position: absolute;
  left: -50px;
}

.appName{
  @media only screen and (max-width: 600px){
    width: 190px;
  }
}

.navMenus{
  margin-left: auto;

  @media only screen and (max-width: 600px) {
    display: none;
  }
}

.navList{
  list-style: none;
  text-align: center;
  padding: 0;
  width: 100%;
  margin: auto;


  @media only screen and (max-width: 600px) {
    margin-top: 8px;
  }

  li{
    display: inline-block;
		margin-right: 4px;
    cursor: pointer;

    @media only screen and (max-width: 600px) {
      border-bottom: 1px solid #fbfbfb;
      display: block;
    }

    a{
      border-radius: 5px;
		  font-size: 18px;
      color: white;
      text-decoration: none;
      display: block;
      padding: 8px;

      &:hover{
        background-color: #902605;
        border: 3px dotted #71230b;
      }
    }
  }
}

.active{
  background-color: #902605;
  border: 3px dotted #71230b;

  @media only screen and (max-width: 600px) {
    margin-bottom: 4px;
    border: 2.5px dotted #71230b;
  }
}
.visible{
  display: block;
  @media only screen and (max-width: 600px) {
    :global{
      animation: fadeInDown .14s linear;
    }
  }
}

.hidden{
  display: none;
}

.home{
  background-color: #fd49119e;
  box-shadow: 0 120px 320px rgba(0, 0, 0, 0.41), 0 15px 12px rgba(0,0,0,0.22);
}

.navToggle{
  display: none;
  outline: none;

  @media only screen and (max-width: 600px){
    display: block;
    background-color: transparent;
    border-radius: 4px;
    border-style: solid;
    border-color: brown;
    color: wheat;
    font-size: 21px;
    width: 45px;
    margin-left: auto;
    height: 36px;
  }
}