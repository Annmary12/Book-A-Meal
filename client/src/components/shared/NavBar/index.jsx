import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import logo from '../../../../assets/img/logo-white.svg';
import styles from './style.scss';

/**
 * @class NavBar
 */
class NavBar extends Component {
  static propTypes = {
    backgroundColor: PropTypes.string
  }
  static defaultProps = {
    backgroundColor: '#e54310'
  }
  state = {
    navMenuStyle: {}
  }
  /**
   * @returns {undefined} - undefined
   */
  toggleMenu() {
    const visible = { display: 'block' };
    const hidden = { display: 'none' };
    this.setState({
      navMenuStyle: this.state.navMenuStyle.display === 'block'
        ? hidden
        : visible
    });
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return (
      <nav className={styles.nav} style={{ backgroundColor: this.props.backgroundColor }}>
        <div className={styles['title-area']}>
          <Link to="/" className={styles.title}>
            <img className={styles.logo} src={logo} width="30" alt="logo" />
            <span className={styles['app-name']}>Book-A-Meal</span>
          </Link>
          <button id="nav-toggle" onClick={() => this.toggleMenu()} href="#">&#9776;</button>
        </div>
        <div className={styles['nav-menus']} style={this.state.navMenuStyle}>
          <ul className={styles['nav-list']}>
            <li>
              <Link to="/login">Login</Link>
            </li>
            <li>
              <Link to="/signup">Sign Up</Link>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default NavBar;
