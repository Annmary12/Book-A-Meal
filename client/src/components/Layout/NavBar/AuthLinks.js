import React, { Fragment } from 'react';
import CustomerLinks from './CustomerLinks.js';
import { Link } from 'react-router-dom';

const AuthLinks = ({ role, pathname }) => {
  //if (role === 'customer') {
    return (
      <CustomerLinks
        pathname={pathname}
      />
    );
  //}
};

export default AuthLinks;
