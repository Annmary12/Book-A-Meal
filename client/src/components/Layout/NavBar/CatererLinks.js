import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import PageLink from './PageLink.js';

const CatererLinks = ({ pathname }) => (
  <Fragment>
    <li>
      <PageLink to="/manage-meals" text="Meals" pathname={pathname} />
    </li>
    <li>
      <PageLink to="/manage-menus" text="Menus" pathname={pathname} />
    </li>
    <li>
      <PageLink to="/manage-orders" text="Orders" pathname={pathname} />
    </li>
  </Fragment>
);

CatererLinks.propTypes = {
  pathname: PropTypes.string.isRequired
};

export default CatererLinks;
