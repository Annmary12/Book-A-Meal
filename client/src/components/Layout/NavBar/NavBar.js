import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import AuthLinks from './AuthLinks.js';
import logo from '../../../../assets/img/logo-white.svg';
import styles from './navbar.scss';

/**
 * @class NavBar
 */
class NavBar extends Component {
  static propTypes = {
    home: PropTypes.bool,
    loggedIn: PropTypes.bool,
    logout: PropTypes.func.isRequired,
    pathname: PropTypes.string.isRequired,
    user: PropTypes.objectOf(PropTypes.string).isRequired
  }
  static defaultProps = {
    home: false,
    loggedIn: false
  }
  state = {
    isNavMenuVisible: false
  }

  /**
   * @returns {undefined} - undefined
   */
  toggleMenu() {
    this.setState({
      isNavMenuVisible: !this.state.isNavMenuVisible
    });
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    const { pathname, user } = this.props;

    return (
      <nav className={classNames({
        [styles.nav]: true,
        [styles.home]: this.props.home
      })}
      >
        <div className={styles.titleArea}>
          <Link to="/" className={styles.title}>
            <img className={styles.logo} src={logo} width="30" alt="logo" />
            <span className={styles.appName}>Book-A-Meal</span>
          </Link>
          <button className={styles.navToggle} onClick={() => this.toggleMenu()} href="#">
            { this.state.isNavMenuVisible ?
              <span style={{ fontSize: 29 }}>&times;</span> :
              <span>&#9776;</span>
            }
          </button>
        </div>
        <div
          className={classNames({
            [styles.navMenus]: true,
            [styles.visible]: this.state.isNavMenuVisible
          })}
        >
          <ul className={styles.navList}>
            {
              this.props.loggedIn ?
                <Fragment>
                  <AuthLinks
                    role={user.role}
                    pathname={pathname}
                  />
                  <li>
                    <Link to="/" onClick={this.props.logout}>Logout</Link>
                  </li>
                </Fragment> :
                <Fragment>
                  <li>
                    <Link to="/login">Login</Link>
                  </li>
                  <li>
                    <Link to="/signup">Sign Up</Link>
                  </li>
                </Fragment>
            }
          </ul>
        </div>
      </nav>
    );
  }
}

export default NavBar;
