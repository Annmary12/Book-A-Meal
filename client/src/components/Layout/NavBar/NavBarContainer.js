import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { logout } from '../../../actions/auth.js';
import NavBar from './NavBar.js';

const mapStateToProps = state => ({
  loggedIn: state.auth.loggedIn,
  user: state.auth.user,
  pathname: state.router.location.pathname
});

const mapDispatchToProps = dispatch => ({
  logout: bindActionCreators(logout, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
