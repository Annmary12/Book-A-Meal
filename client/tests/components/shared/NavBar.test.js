import React from 'react';
import { chai, enzyme } from '../setup';
import NavBar from '../../../src/components/shared/NavBar/index.jsx';

const { shallow } = enzyme;

describe('NavBar Component', () => {
  it('renders properly', () => {
    const tree = shallow(<NavBar />);
    tree.should.matchSnapshot();
  });
});
