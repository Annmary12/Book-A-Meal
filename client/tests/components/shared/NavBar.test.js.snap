// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    backgroundColor="#e54310"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
        >
          <Link
            className={undefined}
            replace={false}
            to="/"
          >
            <img
              alt="logo"
              className={undefined}
              src={Object {}}
              width="30"
            />
            <span
              className={undefined}
            >
              Book-A-Meal
            </span>
          </Link>
          <button
            href="#"
            id="nav-toggle"
            onClick={[Function]}
          >
            ☰
          </button>
        </div>,
        <div
          className={undefined}
          style={Object {}}
        >
          <ul
            className={undefined}
          >
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </li>
          </ul>
        </div>,
      ],
      "className": undefined,
      "style": Object {
        "backgroundColor": "#e54310",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className={undefined}
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                className={undefined}
                src={Object {}}
                width="30"
              />
              <span
                className={undefined}
              >
                Book-A-Meal
              </span>
            </Link>,
            <button
              href="#"
              id="nav-toggle"
              onClick={[Function]}
            >
              ☰
            </button>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className={undefined}
                  src={Object {}}
                  width="30"
                />,
                <span
                  className={undefined}
                >
                  Book-A-Meal
                </span>,
              ],
              "className": undefined,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": undefined,
                  "src": Object {},
                  "width": "30",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Book-A-Meal",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "Book-A-Meal",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "☰",
              "href": "#",
              "id": "nav-toggle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "☰",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className={undefined}
          >
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </li>
          </ul>,
          "className": undefined,
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </li>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "replace": false,
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            <Link
              className={undefined}
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                className={undefined}
                src={Object {}}
                width="30"
              />
              <span
                className={undefined}
              >
                Book-A-Meal
              </span>
            </Link>
            <button
              href="#"
              id="nav-toggle"
              onClick={[Function]}
            >
              ☰
            </button>
          </div>,
          <div
            className={undefined}
            style={Object {}}
          >
            <ul
              className={undefined}
            >
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </li>
            </ul>
          </div>,
        ],
        "className": undefined,
        "style": Object {
          "backgroundColor": "#e54310",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className={undefined}
                replace={false}
                to="/"
              >
                <img
                  alt="logo"
                  className={undefined}
                  src={Object {}}
                  width="30"
                />
                <span
                  className={undefined}
                >
                  Book-A-Meal
                </span>
              </Link>,
              <button
                href="#"
                id="nav-toggle"
                onClick={[Function]}
              >
                ☰
              </button>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt="logo"
                    className={undefined}
                    src={Object {}}
                    width="30"
                  />,
                  <span
                    className={undefined}
                  >
                    Book-A-Meal
                  </span>,
                ],
                "className": undefined,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "logo",
                    "className": undefined,
                    "src": Object {},
                    "width": "30",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Book-A-Meal",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "Book-A-Meal",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "☰",
                "href": "#",
                "id": "nav-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "☰",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className={undefined}
            >
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </li>
            </ul>,
            "className": undefined,
            "style": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </li>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
